### PR TITLE
[CHORE] BlockchainConfig as a case class

### PR DESCRIPTION
**Pull request by ntallar**
# Description

Changes BlockchainConfig from being a trait to a case class.

As seen on a couple of examples here, this will:
- Allow us to use the default copy to only change the parts of the configuration on tests that applies to each
- Follow our current convention of having all our configurations as case classes

### CODE REVIEW

File: src/ets/scala/io/iohk/ethereum/ets/blockchain/ScenarioSetup.scala [hunk 1]
```diff
@@ -43,6 +43,8 @@ object ScenarioSetup {
 
 abstract class ScenarioSetup(_vm: VMImpl, scenario: BlockchainScenario) {
 
+  import BlockchainTestConfig._
```
Comment (mirkoAlic) [inline src/ets/scala/io/iohk/ethereum/ets/blockchain/ScenarioSetup.scala:46]: is this needed?

File: src/test/scala/io/iohk/ethereum/jsonrpc/PersonalServiceSpec.scala [hunk 2]
```diff
@@ -416,37 +416,37 @@ class PersonalServiceSpec extends FlatSpec with Matchers with MockFactory with S
...
     val txValue = 128000
 
-    val blockchainConfig = new BlockchainConfig {
-      override val eip155BlockNumber: BigInt = 12345
-      override val chainId: Byte = 0x03.toByte
+    val blockchainConfig = BlockchainConfig (
+      eip155BlockNumber = 12345,
+      chainId = 0x03.toByte,
 
       //unused
```
Comment (mirkoAlic) [inline src/test/scala/io/iohk/ethereum/jsonrpc/PersonalServiceSpec.scala:423]: all the unused can be removed i guess

File: src/test/scala/io/iohk/ethereum/network/handshaker/EtcHandshakerSpec.scala [hunk 3]
```diff
@@ -172,43 +172,43 @@ class EtcHandshakerSpec extends FlatSpec with Matchers  {
       override val appStateStorage: AppStateStorage = TestSetup.this.storagesInstance.storages.appStateStorage
     }
 
-    override lazy val blockchainConfig = new BlockchainConfig {
+    override lazy val blockchainConfig = BlockchainConfig (
```
Comment (mirkoAlic) [inline src/test/scala/io/iohk/ethereum/network/handshaker/EtcHandshakerSpec.scala:175]: Here you could use the copy given `EphemBlockchainTestSetup` has access to blockchainConfig "default".

**Review by mirkoAlic — APPROVED** [review]
LGTM!

File: src/ets/scala/io/iohk/ethereum/ets/blockchain/ScenarioSetup.scala (hunk 1 above)
Comment (ntallar) [reply src/ets/scala/io/iohk/ethereum/ets/blockchain/ScenarioSetup.scala:46]: It is, as baseBlockchainConfig and withSkippedPoWValidationBlockchainConfig functions use the BlockchainConfig's defined there

**Pull request merged**